src/slopes.jl

```diff
@@ -0,0 +1,198 @@
+# Reference : Dietmar Ratz - An Optimized Interval Slope Arithmetic and its Application
+using IntervalArithmetic, ForwardDiff
+import Base: +, -, *, /, ^, sqrt, exp, log, sin, cos, tan, asin, acos, atan
+import IntervalArithmetic: mid, interval
+
+function slope(f::Function, x::Interval, c::Real)
+    try
+        f(slope_var(x, c)).s
+    catch y
+        if isa(y, MethodError)
+            ForwardDiff.derivative(f, x)
+        end
+    end
+end
+
+struct Slope{T}
+    x::Interval{T}
+    c::Interval{T}
```

Comment:
This should not be an Interval I believe.

Reply:
Won't the `Real` need to bound rounding errors?

Reply:
That depends how it's used. I'm not sure.

Reply:
Interval arithmetic would be used everywhere if `c` is an `Interval` and that should help bound rounding errors, right?